json_encoder.go

```diff
@@ -144,8 +159,12 @@ func (enc *jsonEncoder) AddObject(key string, obj interface{}) error {
 
 // Clone copies the current encoder, including any data already encoded.
 func (enc *jsonEncoder) Clone() encoder {
-	clone := newJSONEncoder()
+	clone := jsonPool.Get().(*jsonEncoder)
```

Comment:
why this change to no longer use `newJSONEncoder` which does the `jsonPool.Get()` + `clone.truncate`?